lobbies.py

```diff
@@ -0,0 +1,185 @@
+    MapVersion("Impossible Bosses BetaV1W", deprecated=True),
+    MapVersion("Impossible Bosses BetaV1V", deprecated=True),
+    MapVersion("Impossible Bosses BetaV1R", deprecated=True),
+    MapVersion("Impossible Bosses BetaV1P", deprecated=True),
+    MapVersion("Impossible Bosses BetaV1C", deprecated=True),
+]
+
+def get_map_version(map_file):
+    for version in KNOWN_VERSIONS:
+        if map_file == version.file_name:
+            return version
+    return None
+
+def get_map_server_nice(server):
+    if server == "usw":
+        return ":flag_us: US"
+    elif server == "eu":
+        return ":flag_eu: EU"
+    elif server == "kr":
+        return ":flag_kr: KR"
+    elif server == "Montreal":
+        return ":flag_ca: Montreal (ENT)"
+    elif server == "New York":
+        return ":flag_us: New York (ENT)"
+    elif server == "France":
+        return ":flag_fr: France (ENT)"
+    elif server == "Amsterdam":
+        return ":flag_nl: Amsterdam (ENT)"
+    return server
+
+class Lobby:
+    def __init__(self, lobby_dict, is_ent):
+        self.is_ent = is_ent
+        self.id = lobby_dict["id"]
+        self.name = lobby_dict["name"]
+        self.map = lobby_dict["map"]
+        self.host = lobby_dict["host"]
+        self.subscribers = []
+
+        if is_ent:
+            self.server = lobby_dict["location"]
+            self.slots_taken = lobby_dict["slots_taken"]
+            self.slots_total = lobby_dict["slots_total"]
+        else:
+            if self.map[-4:] == ".w3x":
+                self.map = self.map[:-4]
+            self.server = lobby_dict["server"]
+            self.slots_taken = lobby_dict["slotsTaken"]
+            self.slots_total = lobby_dict["slotsTotal"]
+
+    def __eq__(self, other):
+        return self.id == other.id
+
+    def __hash__(self):
+        return self.id
+
+    def __str__(self):
+        return "[id={} ent={} name=\"{}\" server={} map=\"{}\" host={} slots={}/{}]".format(
+            self.id, self.is_ent, self.name, self.server, self.map, self.host, self.slots_taken, self.slots_total
+        )
+
+    def get_message_id_key(self):
+        return "lobbymsg{}".format(self.id)
+
+    def is_ib(self):
+        #return self.map.find("Legion") != -1 and self.map.find("TD") != -1 # test
+        #return self.map.find("Uther Party") != -1 # test
+        return self.map.find("Impossible") != -1 and self.map.find("Bosses") != -1
+
+    def is_updated(self, new):
+        return self.name != new.name or self.server != new.server or self.map != new.map or self.host != new.host or self.slots_taken != new.slots_taken or self.slots_total != new.slots_total
+
+    def to_discord_message_info(self, is_open=True):
+        COLOR_CLOSED = discord.Colour(0x8a0808)
+
+        version = get_map_version(self.map)
+        mark = ""
+        message = ""
+        if version is None:
+            mark = ":question:"
+            message = ":warning: *Unknown map version* :warning:"
```

Comment:
All of this file is copy-pasted from `main.py`, except for these lines, where I removed the ugly all-caps `WARNING:` before these messages.